pkg/controller/daemon/controller.go

```diff
@@ -93,9 +93,15 @@ type DaemonSetsController struct {
 
 	// Daemon sets that need to be synced.
 	queue *workqueue.Type
+
+	// Additional conditions that must be satisfied before running a daemon pod on a node
+	conditions []NodeShouldRunDaemonPod
 }
 
-func NewDaemonSetsController(kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc) *DaemonSetsController {
+// NodeShouldRunDaemonPod defines a condition that must return true before a daemon pod will be run a node
+type NodeShouldRunDaemonPod func(node *api.Node, ds *extensions.DaemonSet) bool
+
+func NewDaemonSetsController(kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc, conditions ...NodeShouldRunDaemonPod) *DaemonSetsController {
```

Comment:
I don't like this API.  I'd rather have a single condition so that we aren't choosing between "must match all" and "must match one" sort of choice or dealing with short-circuit logic.